File: consensus/beacon/src/main/java/org/hyperledger/besu/consensus/beacon/containers/beacon_blocks/BeaconBlock.java
package org.hyperledger.besu.consensus.beacon.containers.beacon_blocks;

import org.hyperledger.besu.consensus.beacon.types.Root;
import org.hyperledger.besu.consensus.beacon.types.Slot;
import org.hyperledger.besu.consensus.beacon.types.ValidatorIndex;

public record BeaconBlock(
        Slot slot,
        ValidatorIndex proposerIndex,
        Root parentRoot,
        Root stateRoot,
        BeaconBlockBody body
) {
}

File: beacon/spec/src/main/kotlin/maru/beacon/spec/BeaconBlock.kt
package maru.beacon.spec

import org.apache.tuweni.bytes.Bytes32
import org.hyperledger.besu.consensus.common.bft.ConsensusRoundIdentifier

interface BeaconBlock {
  val consensusRoundIdentifier: ConsensusRoundIdentifier
  val proposer: Validator
  val parentRoot: Bytes32
  val stateRoot: Bytes32
  val beaconBlockBody: BeaconBlockBody
}
